auacm/app/static/js/controllers/EditCompetitionController.js

```diff
@@ -0,0 +1,223 @@
+                (date.getYear() + 1900) + " 10:00";
+    };
+
+    $scope.addProblem = function(problem) {
+        $scope.compProblems.push(problem);
+    };
+    $scope.removeProblem = function(problem) {
+        $scope.compProblems.splice($scope.compProblems.indexOf(problem), 1);
+    };
+    $scope.disableForm = false;
+
+    var getStartTimeSeconds = function() {
+        // get the string from scope and split it into its parts
+        var parts = $scope.startTime.split(' ');
+
+        // The parts consists of the day and time
+        var day = parts[0].split('-');
+        var time = parts[1].split(':');
+
+        // Parse the parts into Numbers so that we can get the time from it.
+        var month = parseInt(day[0]) - 1;
+        var dayOfMonth = parseInt(day[1]);
+        var year = parseInt(day[2]);
+        var hourOfDay = parseInt(time[0]);
+        var minute = parseInt(time[1]);
+
+        // return the seconds of the date that the user entered.
+        return new Date(year, month, dayOfMonth, hourOfDay, minute).valueOf() /
+                1000; // to seconds instead of milliseconds.
+    };
+
+    var getLengthSeconds = function() {
+        var parts = $scope.compLength.split(':');
+        return parseInt(parts[0]) * 3600 + parseInt(parts[1]) * 60;
+    };
+
+    $scope.createCompetition = function() {
+        $scope.disableForm = true;
+        var problems = [];
+        for (var i = 0; i < $scope.compProblems.length; i++) {
+            problems.push({
+                'label': String.fromCharCode("A".charCodeAt(0) + i),
+                'pid': $scope.compProblems[i].pid
+            });
+        }
+        var request = {
+            'name': $scope.compName,
+            'start_time': getStartTimeSeconds(),
+            'length': getLengthSeconds(),
+            'problems': angular.toJson(problems)
+        };
+        $http({
+            method: competition === undefined ? 'POST' : 'PUT',
+            url: competition === undefined ? '/api/competitions' :
+                    '/api/competitions/' + competition.cid,
+            headers: {'Content-Type': 'application/x-www-form-urlencoded'},
+            transformRequest: function(obj) {
+                    var str = [];
+                    for(var p in obj)
+                        str.push(encodeURIComponent(p) + "=" +
+                                encodeURIComponent(obj[p]));
+                    return str.join("&");
+            },
+            data: request
+        }).then(function(response) {
+            $scope.disableForm = false;
+            $scope.compName = '';
+            $scope.compTime = '';
+            $scope.compLength = '';
+            $scope.compProblems = [];
+            $location.path('/competitions/' + response.data.data.cid);
+        }, function(response) {
+            $scope.disableForm = false;
+            console.error(response);
+        });
+    };
+}]);
+
+app.directive('dateFormatValidator', [function() {
```

Comment:
This is cool, how did you find to use directives for form validation? I used something much different when I was doing user creation.